engine/src/main/java/org/terasology/engine/subsystem/rpc/DiscordRPCSubSystem.java

```
@@ -0,0 +1,308 @@
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                    }
+                    continue;
+                }
+
+
+                // Ping to make sure that the RPC is alive
+                if (ready) {
+                    Thread.sleep(1);
+                    lastPing += 1;
+                    if (lastPing >= 5 * 1000) {
+                        ipcClient.sendRichPresence(this.lastRichPresence);
+                        this.lastPing = 0;
+                    }
+                } else { // Reconnect to RPC with tries ( min: 2 seconds, max: 30 seconds )
+                    lastPing = 0;
+                    reconnecting = true;
+                    int timeout = (reconnectTries * 2) * 1000;
+                    getLogger().info("Discord RPC >> Reconnecting... (Timeout: " + timeout + "ms)");
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) {
+                        if (reconnectTries <= 5) {
+                            reconnectTries += 1;
+                        }
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                        Thread.sleep(timeout);
+                    }
+                }
+            } catch (Exception ex) { }
+        }
+    }
+
+    @Override
+    public void initialise(GameEngine engine, Context rootContext) {
+        disableLogger(IPCClient.class);
+        disableLogger(WindowsPipe.class);
+        disableLogger(Pipe.class);
+        Config c = rootContext.get(Config.class);
+        enabled = c.getPlayer().isDiscordPresence();
+        if (!enabled) {
+            return;
+        }
+        try {
+            getLogger().info("Discord RPC >> Connecting...");
+            ipcClient.connect();
+            dontTryAgain = false;
+        } catch (Exception ex) { }
```

Rather than an entirely empty catch block you might want to leave a comment briefly explaining why the exception is being ignored :-)